### ISSUE TITLE
Improved keyboard support

### ISSUE THREAD

**Issue by sean-gugler** [fadden/CiderPress2#25]
Lots of common keyboard shortcuts are already supported, which is great! After using CiderPress II 1.0.2 extensively I find myself wishing for a few more:

* Every menu item should have a keyboard accelerator assigned. (Examples: File/Recent Files, Actions/Edit Attributes)
* In file view, Tab key currently moves between fields. This is redundant with arrow keys; it would be more convenient if (Shift+)Tab changed focus to the (prev/)next UI element.
* Continually hitting Tab key eventually gets "trapped" in the button bar
* F2 to quickly rename a file. Ideally UI would be to edit the name directly in the main view, but for now bringing up the Info dialog with focus on the filename should be enough.
* Press "Enter" to open elected file in Viewer
* In Extract dialog, press "Enter" to invoke the "Select Here" button

**Comment by fadden** [comment]
I like most of these, but anything related to changing focus in WPF can get weird. Just trying to get the initial focus on the file list was causing it to go off into the weeds and disable all actions. It took a while to get it to work as well as it does.

The inter-control tab behavior is probably less of a minefield, but it can be tricky to make it do what you want.

Why F2?

**Comment by fadden** [comment]
> In Extract dialog, press "Enter" to invoke the "Select Here" button

Perhaps it should work the same as double-click: if you do it on a directory it moves into the directory, if you do it on anything else it does Select Here.  (The same dialog is also used to select a mix of files and folders for add/import.)

**Comment by fadden** [comment]
Some new shortcuts...

Main file list:
 - Enter: open file viewer for selected entry.
 - Alt+Enter: open Edit Attributes for selected entry.  (Same as "properties" in Windows Explorer.)

Edit attributes:
 - Alt+D/N/W/R/B/I/L toggle access flag.

File+folder selection dialog:
 - Enter: triggers Select Here.
 - mouse button 4: triggers Nav Up.

**Comment by fadden** [comment]
Main file list:
 - Ctrl+Shift+N: Create Directory (same as in Windows Explorer)
 - Ctrl+Shift+A: Add Files (Ctrl+A was taken by "select all")
 - Ctrl+Shift+1 through Ctrl+Shift+6: open recent file N

This makes Ctrl+Shift+2 flip back and forth between two archives, and Ctrl+Shift+3 rotate through a set of 3.  Higher-numbered shortcuts become increasingly less useful.  They do work when no file is open, so it might be worth displaying the full set on that screen (instead of just the most recent two) to allow Ctrl+W Ctrl+Shift+[#] to be a fast way to switch around.

**Comment by sean-gugler** [comment]
This is awesome! A big help; many thanks.

Still on the wishlist: when focus is on main file list and a directory is selected, "Enter" does not behave as double-click. Instead it tries to open the first file entry found in the list, ignoring the selection, which feels a bit odd.

**Comment by fadden** [comment]
Yeah that feels weird.  The "view files" command expects to get a list of stuff to view.  When only one file is selected, it acts as if all files in the window are selected, so you can use the forward/backward buttons.  You can't really view directories, so it throws those out.  So asking to view a single directory entry gets you the first thing in the window.

The basic problem is that Enter is a shortcut for "view files", rather than being equivalent to double-click, which has special behavior when only one entry is selected.

**Comment by fadden** [comment]
Updated in v1.0.5-dev1.  You should be able to navigate through the directory hierarchy in "single directory" view mode using the keyboard.
